pulsar-broker/src/main/java/org/apache/pulsar/broker/systopic/SystemTopicClient.java

```diff
@@ -167,8 +167,11 @@
         SystemTopicClient<T> getSystemTopic();
     }
 
+    /**
+     * A topic is a system topic if and only if its local name has the defined prefix.
+     */
     static boolean isSystemTopic(TopicName topicName) {
-        return EventsTopicNames.NAMESPACE_EVENTS_LOCAL_NAME.equals(topicName.getLocalName());
+        return topicName.getLocalName().startsWith(EventsTopicNames.SYSTEM_TOPIC_LOCAL_NAME_PREFIX);
```

Comment:
The one major consequence of this implementation is the all topics created or existing with `__` would be considered system topics. Perhaps that is too much of a breaking change.

I updated it this way because the old implementation ignored the `TRANSACTION_BUFFER_SNAPSHOT` topic, which is a system topic. It seems easier to maintain if we just follow the paradigm instead of adding every new system topic name as we create new ones.

Reply:
i don't think `__` would be considered system topics is a good idea. It will cause version incompatibility.

Reply:
That is a great point. It does seem like there is already a bit of a trend where system topics begin with `__`, but you're right that it would need to be introduced more intentionally than this PR.

I'll look at changing this PR accordingly.